### PR TITLE
Don't use all caps and distinguish between upper bound and maximum

### PR DESCRIPTION
**Pull request by mkver**

### CODE REVIEW

**Comment by retokromer** [comment]
Modify the title of this PR?

**Comment by mkver** [comment]
I added a commit that better distinguishes between maximum and upper bound; it modifies the same paragraph that the all caps commit modifies and is therefore part of the same PR (otherwise there would be merge conflicts).

File: specification.markdown [hunk 1]
```diff
@@ -423,15 +423,15 @@ The `id` attribute is REQUIRED.
 
 #### minOccurs
 
-An integer expressing the minimum number of occurrences of this `EBML Element` within its `Parent Element`. The `minOccurs` value MUST be equal to the `EBMLMinOccurrence` value of the `path`.
+An integer expressing the minimum permitted number of occurrences of this `EBML Element` within its `Parent Element`. The `minOccurs` value MUST be equal to the `EBMLMinOccurrence` value of the `path`.
```
Comment (robUx4) [inline specification.markdown]: Shouldn't `permitted` be before `within` ? :
> An integer expressing the minimum number of occurrences of this `EBML Element` permitted within its `Parent Element`.

File: specification.markdown [hunk 2]
```diff
@@ -423,15 +423,15 @@ The `id` attribute is REQUIRED.
...
 
 The `minOccurs` attribute is OPTIONAL. If the `minOccurs` attribute is not present then that `EBML Element` has a `minOccurs` value of 0.
 
 #### maxOccurs
 
-An integer expressing the maximum number of occurrences of this `EBML Element` within its `Parent Element`. The `maxOccurs` value MUST be equal to the `EBMLMaxOccurrence` value of the `path`.
+An integer expressing the maximum permitted number of occurrences of this `EBML Element` within its `Parent Element`. The `maxOccurs` value MUST be equal to the `EBMLMaxOccurrence` value of the `path`.
```
Comment (robUx4) [inline specification.markdown]: Shouldn't `permitted` be before `within` ? :

> An integer expressing the maximum number of occurrences of this `EBML Element` permitted within its `Parent Element`.

File: specification.markdown [hunk 3]
```diff
@@ -423,15 +423,15 @@ The `id` attribute is REQUIRED.
...
 
-The `maxOccurs` attribute is OPTIONAL. If the `maxOccurs` attribute is not present then that `EBML Element` has no maximum occurrence, similar to `unbounded` in the XML world.
+The `maxOccurs` attribute is OPTIONAL. If the `maxOccurs` attribute is not present then no further restrictions are imposed on the number of occurences of this `EBML Element` besides the restriction imposed by `minOccurs`. This is similar to `unbounded` in the XML world.
```
Comment (robUx4) [inline specification.markdown]: Referencing `minOccurs` here add more complexity to understand what happens if `maxOccurs` is not present. Saying "there's no restriction on that number except if there is one" doesn't help IMO. I prefer the original text.

File: specification.markdown [hunk 4]
```diff
@@ -659,7 +659,7 @@ default: 4
 
 type: Unsigned Integer
 
-description: The `EBMLMaxIDLength Element` stores the maximum length in octets of the `Element IDs` to be found within the `EBML Body`. An `EBMLMaxIDLength Element` value of four is RECOMMENDED, though larger values are allowed.
+description: The `EBMLMaxIDLength Element` stores the maximum permitted length in octets of the `Element IDs` to be found within the `EBML Body`. An `EBMLMaxIDLength Element` value of four is RECOMMENDED, though larger values are allowed.
```
Comment (robUx4) [inline specification.markdown]: Same remark on the location of `permitted` which sounds better to me as:
> the maximum length in octets of the `Element IDs` permitted to be found within the `EBML Body`

File: specification.markdown [hunk 5]
```diff
@@ -679,7 +679,7 @@ default: 8
 
 type: Unsigned Integer
 
-description: The `EBMLMaxSizeLength Element` stores the maximum length in octets of the expression of all `Element Data Sizes` to be found within the `EBML Body`. To be clear the `EBMLMaxSizeLength Element` documents the maximum 'length' of all `Element Data Size` expressions within the `EBML Body` and not the maximum 'value' of all `Element Data Size` expressions within the `EBML Body`. `EBML Elements` that have an `Element Data Size` expression which is larger in octets than what is expressed by `EBMLMaxSizeLength ELEMENT` are invalid.
+description: The `EBMLMaxSizeLength Element` stores the maximum permitted length in octets of the expressions of all `Element Data Sizes` to be found within the `EBML Body`. The `EBMLMaxSizeLength Element` documents an upper bound for the 'length' of all `Element Data Size` expressions within the `EBML Body` and not an upper bound for the 'value' of all `Element Data Size` expressions within the `EBML Body`. `EBML Elements` that have an `Element Data Size` expression which is larger in octets than what is expressed by `EBMLMaxSizeLength Element` are invalid.
```
Comment (robUx4) [inline specification.markdown]: 👍

File: specification.markdown [hunk 6]
```diff
@@ -407,9 +407,9 @@ The starting `PathDelimiter` of the path corresponds to the root of the `EBML Do
...
+The `EBMLMaxOccurrence` represents the maximum permitted number of occurrences of this `EBML Element` within its `Parent Element`. Each instance of the `Parent Element` MUST contain at most this many instances of this `EBML Element`. If the `EBML Element` has an empty `EBMLParentPath` then `EBMLMaxOccurrence` refers to constraints on the occurrence of the `EBML Element` within the `EBML Document`. If `EBMLMaxOccurrence` is not present then there is no upper bound for the permitted number of occurences of this `EBML Element` within its `Parent Element` resp. within the `EBML Document` depending on whether the `EBMLParentPath` of the `EBML Element` is empty or not. The semantic meaning of `EBMLMaxOccurrence` within an `EBML Schema path` is analogous to the meaning of `maxOccurs` within an `XML Schema`.
```
Comment (robUx4) [inline specification.markdown]: 👍

**Comment by robUx4** [comment]
I think the added word `permitted` doesn't add much to the text. IMO it's clear that the whole document defines what is allowed or what is not.

The notion of  `upper bound` seems to make things clearer.

File: specification.markdown (hunk 3 above)
Comment (robUx4) [reply specification.markdown]: The rest of the changes I can live with, but I'd rather have this change removed or simplified.

**Comment by robUx4** [comment]
I merged these changes except for the `maxOccurs` extra text. Rebasing this branch on master should just keep this change.

**Comment by robUx4** [comment]
Can you rebase your branch or close this one and create a new PR for the `maxOccurs` change ?

**Comment by robUx4** [comment]
Closing as won't fix/outdated.

**Pull request closed without merging**